### ISSUE TITLE
Linter fails to identify missing whitespace between string literal and THEN

### ISSUE THREAD

**Issue by jmskarda** [sqlfluff/sqlfluff#3996]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The linter fails to identify missing whitespace between a string literal and "THEN" in a CASE...WHEN...THEN...END statement.

### Expected Behaviour

The linter should be able to identify the issue as a violation of rule L048.

### Observed Behaviour

Linter does not identify the missing whitespace issue.

### How to reproduce

```sql
SELECT
  CASE
    WHEN field1 ='a' THEN 1 -- Linter correctly identifies violation of L048
    WHEN field1 = 'b'THEN 2 -- Linter fails to identify violation of L048
  END AS alias1
FROM
  foo
;
```

### Dialect

snowflake

### Version

sqlfluff, version 1.3.2
python, version 3.7.6

### Configuration

[sqlfluff]
rules = core

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by joaostorrer** [comment]
I tested with sqlfluff 3.3.1 and is correetcly identifying both cases now.

**Comment by keraion** [comment]
Fixed by #3824
